src/Common/src/TypeSystem/NativeFormat/NativeFormatType.cs

```diff
@@ -458,8 +458,7 @@ public override MethodDesc GetFinalizer()
                 return null;
             }
 
-            // TODO: Better exception type. Should be: "CoreLib doesn't have a required thing in it".
-            throw new NotImplementedException();
+            throw new TypeLoadException(this.GetFullName());
```

Comment:
This should be `throw new TypeSystemException.TypeLoadException` so that we can catch it from places.